frontend/src/scenes/dashboard/dashboardLogic.js

```diff
@@ -105,6 +105,24 @@ export const dashboardLogic = kea({
             [dashboardsModel.actions.updateDashboardItem]: (state, { item }) => {
                 return { ...state, items: state.items.map((i) => (i.id === item.id ? item : i)) }
             },
+            [dashboardsModel.actions.updateDashboardRefreshStatus]: (state, { id, refreshing, last_refresh }) => {
+                // If not a dashboard item, don't do anything.
+                if (!id) {
+                    return state
+                }
+                return {
+                    ...state,
+                    items: state.items.map((i) =>
+                        i.id === id
+                            ? {
+                                  ...i,
+                                  ...(refreshing == null ? { refreshing } : {}),
```

Comment:
Hmm I think it'll actually be less confusing if I change the types here so that none of these parameters can be undefined. I think both refreshing should be typed to `boolean | null` while last_refresh is `string | null`.